### PR TITLE
test: run tests on webkit for webRTC

### PR DESCRIPTION
**Pull request by maschad**

### CODE REVIEW

File: packages/transport-webrtc/package.json [hunk 1]
```diff
@@ -38,18 +38,20 @@
     "test": "aegir test -t browser",
     "test:chrome": "aegir test -t browser --cov",
     "test:firefox": "aegir test -t browser -- --browser firefox",
+    "test:webkit": "aegir test -t browser -- --browser webkit",
     "lint": "aegir lint",
     "lint:fix": "aegir lint --fix",
     "clean": "aegir clean",
-    "dep-check": "aegir dep-check"
+    "dep-check": "aegir dep-check -i protons"
```
Comment (achingbrain) [inline packages/transport-webrtc/package.json:45]: This is already in the [default ignore list](https://github.com/ipfs/aegir/blob/369453f7543488d43e72721c48320d6d8d59b705/src/dependency-check.js#L13)

```suggestion
    "dep-check": "aegir dep-check"
```

File: packages/transport-webrtc/package.json [hunk 2]
```diff
@@ -68,17 +70,16 @@
...
+    "@libp2p/peer-id-factory": "^2.0.4",
+    "@libp2p/websockets": "^6.0.3",
     "@types/sinon": "^10.0.15",
     "aegir": "^39.0.10",
     "delay": "^6.0.0",
     "it-length": "^3.0.2",
     "it-map": "^3.0.3",
     "it-pair": "^2.0.6",
-    "libp2p": "^0.45.0",
+    "libp2p": "^0.45.9",
```
Comment (achingbrain) [inline packages/transport-webrtc/package.json:82]: Not necessary as this is a monorepo sibling and this change will be undone by `aegir check-project`.

```suggestion
    "libp2p": "^0.45.0",
```

File: packages/transport-webrtc/package.json [hunk 3]
```diff
@@ -68,17 +70,16 @@
     "uint8arrays": "^4.0.4"
   },
   "devDependencies": {
-    "@chainsafe/libp2p-yamux": "^4.0.1",
-    "@libp2p/interface-compliance-tests": "^3.0.0",
-    "@libp2p/peer-id-factory": "^2.0.0",
-    "@libp2p/websockets": "^6.0.0",
+    "@chainsafe/libp2p-yamux": "^4.0.2",
+    "@libp2p/peer-id-factory": "^2.0.4",
+    "@libp2p/websockets": "^6.0.3",
```
Comment (achingbrain) [inline packages/transport-webrtc/package.json:75]: Not necessary as these are monorepo siblings and this change will be undone by `aegir check-project`.

```suggestion
    "@libp2p/peer-id-factory": "^2.0.0",
    "@libp2p/websockets": "^6.0.0",
```

File: packages/transport-webrtc/test/basics.spec.ts [hunk 4]
```diff
@@ -85,15 +85,16 @@ describe('basics', () => {
...
-  afterEach(async () => {
-    if (localNode != null) {
-      await localNode.stop()
-    }
-
-    if (remoteNode != null) {
-      await remoteNode.stop()
-    }
-  })
+  // TODO: Streams are not closing gracefully, re-introduce after https://github.com/libp2p/js-libp2p/issues/1793 is addressed
```
Comment (achingbrain) [inline packages/transport-webrtc/test/basics.spec.ts:88]: The issue in #1793 does not prevent nodes from stopping.  It prevents stream data from being sent to the remote before the stream is closed.

Why does that mean these lines need to be commented out?

**Review by achingbrain — CHANGES_REQUESTED** [review]

**Review by achingbrain — CHANGES_REQUESTED** [review]
This needs a `test:webkit` script added to the `package.json` in the monorepo root as well, otherwise the webkit tests will not be run.

**Comment by maschad** [comment]
it seems on webkit that the `RTCDataChannel` doesn't send the `arraybuffer` binaryType the same way that other browsers do, which results [in some failures when closing a connection](https://github.com/libp2p/js-libp2p/actions/runs/5455886352/jobs/9928054478?pr=1832#step:6:562), I a bit unsure how to proceed as the only way I have gotten this to [work is to add a` delay(10)` or greater](https://github.com/libp2p/js-libp2p/pull/1832/files#diff-05b91eb525d98768ddd0ab5a8d129630baf7b747a59df7ce6418fefb9c3c4f22R158), which is obviously not a fix.

**Comment by maschad** [comment]
Currently blocked per https://github.com/libp2p/js-libp2p-webrtc/pull/177#issuecomment-1589529908

**Comment by maschad** [comment]
Closing this until https://github.com/libp2p/js-libp2p/issues/2007 is resolved

**Pull request closed without merging**